File: CLIENT/src/app/models/group.ts
export interface Group{
    name: string; 
    connections: Connection[];
    unreadMessages: boolean;
    lastMessageSender: string;
    lastMessageContent: string;
    username1: string;
    username2: string;
    user1Photo: string;
    user2Photo: string;
}

interface Connection{
    connectionId: string;
    username: string;
}

File: API/Entities/Group.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities
{
    public class Group
    {
        public Group()
        {
        }

        public Group(string? name)
        {
            Name = name;
        }

        [Key]
        public string? Name { get; set; }
        public ICollection<Connection>? Connections { get; set; } = new List<Connection>();
        public bool UnreadMessages { get; set; }
        public string? LastMessageSender { get; set; }
        public string? LastMessageContent { get; set; }
        public string? Username1 { get; set; }
        public string? Username2 { get; set; }
    }
}